arch/eventloop_posix_udp.c

```diff
@@ -335,11 +336,15 @@ setConnectionConfigurationProperties(UA_FD socket,
                    UA_String_equal(&prop->key.name, &listenPortParam)) {
             /* ignore, required args are handled elsewhere explicitly */
         } else {
-            char keyname[prop->key.name.length + 1];
-            strncpy(keyname, (char*)prop->key.name.data, prop->key.name.length);
-            keyname[prop->key.name.length] = '\0';
+            char keyName[UA_MAXPARAMSTR_LENGTH];
+            size_t nonNullTerminatedLength = prop->key.name.length;
+            if (prop->key.name.length > UA_MAXPARAMSTR_LENGTH - 1) {
+                nonNullTerminatedLength = UA_MAXPARAMSTR_LENGTH - 1;
+            }
+            strncpy(keyName, (char*)prop->key.name.data, nonNullTerminatedLength);
+            keyName[nonNullTerminatedLength] = '\0';
             UA_LOG_WARNING(logger, UA_LOGCATEGORY_SERVER,
-                           "PubSub Connection creation. Unknown connection parameter: '%s'.", keyname);
+                           "PubSub Connection creation. Unknown connection parameter: '%s'.", keyName);
```

Comment:
You can use the following idiom to print non-null terminated strings.
That might save a bit of accidental complexity.

```c
printf("my string %.*s", (int)str->length, (char*)str->data);
```

Reply:
Perfect thanks!